### PR TITLE
llvm: add upstream changes Linuxbrew

### PR DESCRIPTION
**Pull request by iMichka**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/linuxbrew-core/blob/master/CONTRIBUTING.md)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/linuxbrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?
- [ ] Have you included the output of `brew gist-logs <formula>` of the build failure if your PR fixes a build failure. Please quote the exact error message.

-----

### CODE REVIEW

**Comment by Bo98** [comment]
Add this to the stable block:

```rb
unless OS.mac?
  patch :p2 do
    url "https://github.com/llvm/llvm-project/commit/7f5fe30a150e7e87d3fbe4da4ab0e76ec38b40b9.patch?full_index=1"
    sha256 "9ed85d2b00d0b70c628a5d1256d87808d944532fe8c592516577a4f8906a042c"
  end
end
```

Add this to the `unless OS.mac?` dependencies (not "required" but leaving it out will mean some OpenMP features are disabled):

```rb
depends_on "pkg-config" => :build
```

`LLVM_LINK_LLVM_DYLIB` is the difference between a 1GB install size and a 4GB+ install size. I'll leave that up to you to decide whether to build statically or via shared libraries.

**Comment by Bo98** [comment]
If you want your clang patch to apply on `--HEAD`:

```rb
head do
  url "https://github.com/llvm/llvm-project.git"

  unless OS.mac?
      patch do
        url "https://gist.githubusercontent.com/iMichka/9ac8e228679a85210e11e59d029217c1/raw/e50e47df860201589e6f43e9f8e9a4fc8d8a972b/clang9?full_index=1"
        sha256 "65cf0dd9fdce510e74648e5c230de3e253492b8f6793a89534becdb13e488d0c"
        directory "clang"
      end
    end
end
```

However I've not tested if it applies cleanly or not.

File: Formula/llvm.rb [hunk 1]
```diff
@@ -18,10 +19,15 @@ class Llvm < Formula
           url "https://gist.githubusercontent.com/iMichka/9ac8e228679a85210e11e59d029217c1/raw/e50e47df860201589e6f43e9f8e9a4fc8d8a972b/clang9?full_index=1"
           sha256 "65cf0dd9fdce510e74648e5c230de3e253492b8f6793a89534becdb13e488d0c"
         end
+
+        patch :p2 do
+          url "https://github.com/llvm/llvm-project/commit/7f5fe30a150e7e87d3fbe4da4ab0e76ec38b40b9.patch?full_index=1"
+          sha256 "9ed85d2b00d0b70c628a5d1256d87808d944532fe8c592516577a4f8906a042c"
+        end
```
Comment (Bo98) [inline Formula/llvm.rb:26]: This should be outside the `clang` resource.

**Comment by iMichka** [comment]
Thanks for the patch @Bo89 !

FYI the bottle is now 231.51 MB vs 859.34 MB before. We do use the strip command on the files to trim down the size.

**Pull request merged**